contextproject-ms3/src/main/java/contextproject/audio/ComplexOnsetDetector.java

```diff
@@ -0,0 +1,233 @@
+ * The additions will be commented and are recognized by the comments within '-MS3-'.
+ * -MS3-
+ * 
+ * A complex Domain Method onset detection function
+ * 
+ * Christopher Duxbury, Mike E. Davies, and Mark B. Sandler. Complex domain
+ * onset detection for musical signals. In Proceedings of the Digital Audio
+ * Effects Conference, DAFx-03, pages 90-93, London, UK, 2003
+ * 
+ * The implementation is a translation of onset.c from Aubio, Copyright (C)
+ * 2003-2009 Paul Brossier <piem@aubio.org>
+ * 
+ * @author Joren Six
+ * @author Paul Brossiers
+ */
+public class ComplexOnsetDetector implements AudioProcessor, OnsetDetector{
+  
+  
+  /**
+   * The threshold to define silence, in dbSPL.
+   */
+  private final double silenceThreshold;
+  
+  /**
+   * The minimum IOI (inter onset interval), in seconds.
+   */
+  private final double minimumInterOnsetInterval;
+  
+  /**
+   * The last detected onset, in seconds.
+   */
+  private double lastOnset;
+  
+  private final PeakPicker peakPicker;
+  
+  private OnsetHandler handler;
+  
+  
+  /**
+   * To calculate the FFT.
+   */
+  private final FFT fft;
+  
+  /**
+   * Previous phase vector, one frame behind
+   */
+  private final float[] theta1;
+  /**
+   * Previous phase vector, two frames behind
+   */
+  private final float[] theta2;
+  
+  /**
+   * Previous norm (power, magnitude) vector
+   */
+  private final float[] oldmag;
+  
+  /**
+   * Current onset detection measure vector 
+   */
+  private final float[] dev1;
+  
+  /**
+   * -MS3-
+   * First onset.
+   * -MS3-
+   */
+  private double firstOnset;
+  
+  /**
+   * -MS3-
+   * Boolean to detect the first onset.
+   * Set to false if it is detected.
+   * -MS3-
+   */
+  private boolean firstBoolean = true;
+  
+  /**
+   * 
+   * @param fftSize The size of the fft to take (e.g. 512)
+   * @param peakThreshold A threshold used for peak picking. Values between 0.1 and 0.8. Default is 0.3, if too many onsets are detected adjust to 0.4 or 0.5.
+   * @param silenceThreshold The threshold that defines when a buffer is silent. Default is -70dBSPL. -90 is also used.
+   * @param minimumInterOnsetInterval The minimum inter-onset-interval in seconds. When two onsets are detected within this interval the last one does not count. Default is 0.004 seconds.
+   */
+  public ComplexOnsetDetector(int fftSize,double peakThreshold,double minimumInterOnsetInterval,double silenceThreshold){
+    fft = new FFT(fftSize,new HannWindow());
+    this.silenceThreshold = silenceThreshold;
+    this.minimumInterOnsetInterval = minimumInterOnsetInterval;
+    
+    peakPicker = new PeakPicker(peakThreshold);
+    
+    int rsize = fftSize/2+1;
+    oldmag = new float[rsize];
+    dev1 = new float[rsize];
+    theta1 = new float[rsize];
+    theta2 = new float[rsize];
+    
+    handler = new PrintOnsetHandler();
+  }
+  
+  public ComplexOnsetDetector(int fftSize){
```

Comment:
I know this is not your code but for the main picture it would be really nice when a javadoc comment is provided for these methods too.